UnityProject/Assets/Scripts/Core/UniversalObjectPhysics.cs

```diff
@@ -1060,16 +1077,8 @@ public void PullSet(UniversalObjectPhysics toPull, bool byClient, bool synced =
 		if (float.IsNaN(nairTime) == false || float.IsNaN(InSlideTime) == false)
 		{
 			worldDirection.Normalize();
-			if (newtonianMovement.magnitude > 0)
-			{
-				//Logger.LogError("  momentum boost");
-			}
 
-			newtonianMovement += worldDirection * speed;
-			if (newtonianMovement.magnitude > MAX_SPEED)
-			{
-				newtonianMovement *= (MAX_SPEED / newtonianMovement.magnitude);
-			}
+			SetNewtonianMovement = newtonianMovement + worldDirection * speed;
```

Comment:
Why not use the new property here too? It could then be simplified to just:
```suggestion
			SetNewtonianMovement += worldDirection * speed;
```